### ISSUE TITLE
[BUG] The Locus 0.14.2 (5343) is not installed from the F-Droid

### ISSUE THREAD

**Issue by Korb** [Myzel394/locus#99]
## Device
Android
OS version: Funtouch OS 13 Global
Model: V2153 (Vivo T1 4G)
Hardware version: MP_0.1
Build number: PD2201OF_EX_A_13.1.15.4.W20.V000L1
Baseband version: MPSS.HA.1.1.c2-00012-KD_ALL_PACK-1.6347.102
Kernel version: 4.19.157-perf+. 1 Tue Jul 4 17:48:44 CST 2023
Compile time: 4 Jul 2023 12:28:54
Android security update: 1 July 2023
Google Play system update: 1 July 2023

## Describe the bug
The app is not installed from the F-Droid app store.

## To Reproduce
1. Search Locus at F-Droid mobile app
2. Tap Install

## Screenshots
![Screenshot_20230830_220932_org fdroid fdroid](https://github.com/Myzel394/locus/assets/3509797/92170848-5362-4283-b7c8-228729c1c604)

## Additional context
Locus (Share your location end-to-end encrypted on decentralized servers using Nostr)
https://f-droid.org/packages/app.myzel394.locus/

## Priority

* [ ] Low
* [ ] Medium
* [ ] High
* [X] Urgent
 
## App Version
> Version 0.14.2 (5343)
> Added on 22/08/2023
> Repository: F-Droid
> Size: 28.4 MiB
> Android: 7.0+
> Requires: arm64-v8a

**Comment by Myzel394** [comment]
Yes I can confirm that it is indeed missing on IzzyOnDroid. However, since then there's a new version available under https://apt.izzysoft.de/fdroid/repo/app.myzel394.locus_5352.apk. Can you update to this version @Korb ? 

I'm still curious, why version code 34 has not been uploaded to Izzy. @IzzySoft do you know why that was the case?

**Comment by IzzySoft** [comment]
Given the size of the APK, my repo only keeps 1 version (more than 1 would exceed the per-app limit of 30 MB). So that APK probably was there, but then a new version was released and thus the APK was replaced by a newer one. Now @Korb most likely had not yet the latest index, but the one from before – so that index still pointed to the APK which was no longer there.

TL;DR: Works as designed. And happens in the short interval between the repo being updated at the server, and the index being updated on the client. Can only be avoided when more than one version is kept.

**Comment by Korb** [comment]
After forcing the repositories update in the F-Droid app, the Locus page now shows only one version:

> 0.14.3 (5352) Suggested
> Added on 30/08/2023
> Repository: IzzyOnDroid F-Droid Repo
> Size: 26.9 MiB
> Android: 7.0+
> Requires: armeabi-v7a

It installed without problems.

**Comment by IzzySoft** [comment]
> After forcing the repositories update in the F-Droid app, the Locus page now shows only one version

Which in this case is the expected result. So what I forgot to mention above: Whenever you get a 404 while trying to install/update an app, pull to refresh the index and try again is the proper solution (unless the app in question was removed entirely for some reason, which is a different issue). Glad it solved!
